### DISCUSSION TITLE
Use enum (that is the `@id`) as one-to-many relation (`v4.4.0`)

### DISCUSSION THREAD

**Discussion by bennettdams** [prisma/prisma#15695] [Q&A] [answered]
### Update

This way of using enums works fine, I just had a weird setup.
In my case, I had the schema with the models in place already, but I used `string` instead of the described enum before.
Me changing my schema to the enum-variant obviously created some type errors in the app code (and also the Prisma error shown below). When I fixed the app type errors after migrating to enum, the Prisma error also went away - no idea why.

---
### Original post

I tried using an enum (that is the `@id` of one field) as a relation to another model. The result was a type error in the prisma base types:

```
  node_modules/.prisma/client/index.d.ts:11046:5

    11046     musicPlatform?: MusicPlatformUpdateOneRequiredWithoutSongLinkNestedInput
              ~~~~~~~~~~~~~~~~
    The expected type comes from property 'musicPlatform' which is declared here on type '(Without<SongLinkUpdateInput, SongLinkUncheckedUpdateInput> & SongLinkUncheckedUpdateInput) | (Without<...> & SongLinkUpdateInput)'
```

Before I create an issue with a reproduction, is this expected work in the first place?

Schema:

```prisma
enum MusicPlatformId {
  SPOTIFY
  SOUNDCLOUD
}

model MusicPlatform {
  musicPlatformId MusicPlatformId @id
  name            String
  // song links relation
  SongLink        SongLink[]      @relation("SongLink_MusicPlatform")
}

model SongLink {
  songLinkId      String          @id @default(cuid())
  address         String
  // music platform relation
  musicPlatform   MusicPlatform   @relation("SongLink_MusicPlatform", fields: [musicPlatformId], references: [musicPlatformId])
  musicPlatformId MusicPlatformId
}






```

**Comment by austincrim** [comment, marked as answer]
Hey @bennettdams 👋,

Glad it got resolved. If you see any other weird behavior, don't hesitate to open another discussion 👍.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
